src/nerve_tools/configure_nerve.py

```diff
@@ -182,13 +186,16 @@ def get_envoy_listeners(admin_port: int) -> Mapping[str, int]:
 def _get_envoy_service_info(
     service_name: str,
     service_info: ServiceInfo,
-    envoy_listeners: Mapping[str, int],
+    envoy_listeners: Mapping[int, int],
 ) -> Optional[ServiceInfo]:
     envoy_service_info: Optional[ServiceInfo] = None
-    envoy_key = f"{service_name}.{service_info['port']}"
-    if envoy_listeners and envoy_key in envoy_listeners:
+    local_host_port = service_info['port']
+    # If this service's local host port is being routed to from an Envoy ingress port,
+    # then output nerve configs so that this service will be healthchecked through
+    # the Envoy ingress port. This requires setting the healthcheck Host header too.
```

Comment:
Worth noting _why_ we need the healthcheck host header? And putting a warning about potential races?